WordPress/Classes/ViewRelated/Post/PostListViewController.m

```diff
@@ -0,0 +1,1319 @@
+        navController.restorationClass = [WPPostViewController class];
+    } else {
+        WPLegacyEditPostViewController *editPostViewController = [[WPLegacyEditPostViewController alloc] initWithDraftForLastUsedBlog];
+        navController = [[UINavigationController alloc] initWithRootViewController:editPostViewController];
+        navController.restorationIdentifier = WPLegacyEditorNavigationRestorationID;
+        navController.restorationClass = [WPLegacyEditPostViewController class];
+    }
+
+    [navController setToolbarHidden:NO]; // Fixes incorrect toolbar animation.
+    navController.modalPresentationStyle = UIModalPresentationFullScreen;
+
+    [self presentViewController:navController animated:YES completion:nil];
+
+    [WPAnalytics track:WPAnalyticsStatEditorCreatedPost withProperties:@{ @"tap_source": @"posts_view" }];
+}
+
+- (void)publishPost:(AbstractPost *)apost
+{
+    apost.status = PostStatusPublish;
+    apost.dateCreated = [NSDate date];
+    PostService *postService = [[PostService alloc] initWithManagedObjectContext:[[ContextManager sharedInstance] mainContext]];
+    [postService uploadPost:apost
+                    success:nil
+                    failure:^(NSError *error) {
+                        if([error code] == 403) {
+                            [self promptForPasswordWithMessage:nil];
+                        } else {
+                            [WPError showXMLRPCErrorAlert:error];
+                        }
+                        [self syncItemsWithUserInteraction:NO];
+                    }];
+}
+
+- (void)editPost:(AbstractPost *)apost
+{
+    if ([WPPostViewController isNewEditorEnabled]) {
+        WPPostViewController *postViewController = [[WPPostViewController alloc] initWithPost:apost
+                                                                                         mode:kWPPostViewControllerModePreview];
+        postViewController.hidesBottomBarWhenPushed = YES;
+        [self.navigationController pushViewController:postViewController animated:YES];
+    } else {
+        // In legacy mode, view means edit
+        WPLegacyEditPostViewController *editPostViewController = [[WPLegacyEditPostViewController alloc] initWithPost:apost];
+        UINavigationController *navController = [[UINavigationController alloc] initWithRootViewController:editPostViewController];
+        [navController setToolbarHidden:NO]; // Fixes incorrect toolbar animation.
+        navController.modalPresentationStyle = UIModalPresentationFullScreen;
+        navController.restorationIdentifier = WPLegacyEditorNavigationRestorationID;
+        navController.restorationClass = [WPLegacyEditPostViewController class];
+
+        [self presentViewController:navController animated:YES completion:nil];
+    }
+}
+
+- (void)viewPost:(AbstractPost *)apost
+{
+    PostPreviewViewController *controller = [[PostPreviewViewController alloc] initWithPost:apost shouldHideStatusBar:NO];
```

Comment:
Minor thing, but the "View" button launches a controller with "Preview" in the title. We might want to tweak that